src/Controls/src/Xaml/SimplifyOnPlatformVisitor.cs

```diff
@@ -0,0 +1,77 @@
+﻿#nullable disable
+
+namespace Microsoft.Maui.Controls.Xaml
+{
+	class SimplifyOnPlatformVisitor : IXamlNodeVisitor
+	{
+		public SimplifyOnPlatformVisitor(string targetFramework)
+		{
+			TargetFramework = targetFramework;
+		}
+
+		public string TargetFramework { get; }
+
+		public TreeVisitingMode VisitingMode => TreeVisitingMode.BottomUp;
+		public bool StopOnDataTemplate => false;
+		public bool VisitNodeOnDataTemplate => true;
+		public bool StopOnResourceDictionary => false;
+		public bool IsResourceDictionary(ElementNode node) => false;
+		public bool SkipChildren(INode node, INode parentNode) => false;
+
+		public void Visit(ValueNode node, INode parentNode)
+		{
+		}
+
+		public void Visit(MarkupNode node, INode parentNode)
+		{
+		}
+
+		public void Visit(ElementNode node, INode parentNode)
+		{
+			if (node.XmlType.Name != nameof(OnPlatformExtension) || node.XmlType.NamespaceUri != XamlParser.MauiUri)
+				return;
+			if (string.IsNullOrEmpty(TargetFramework))
+				return;
+
+			string target = null;
+			if (TargetFramework.Contains("-android"))
```

Comment:
`TargetFramework` doesn't change for the lifetime of this object. Does it make sense to do this switch every time?

Reply:
despite being in the Xaml project, this code is only ever executed by the compiler, and we're usually ok to sacrifice a few ms over code simplicity and readability. I'm 100% sure there's an alternative code that is as readable, as debuggable and as simple, but I'm not sure it will ever save the time we currently spend discussing it :)